src/modules/Game.class.js

```diff
@@ -50,19 +223,135 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+
+  getStatus() {
+    let emptyField = true;
+
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 4; j++) {
+        if (this.field[i][j] === 2048) {
+          document.querySelector('.message-win').style.display = 'block';
+          document.removeEventListener('keydown', this.keyMove);
+
+          return 'win';
+        }
+
+        if (this.field[i][j] !== 0) {
+          emptyField = false;
+        }
+      }
+    }
+
+    if (emptyField) {
+      return 'idle';
+    }
+
+    if (this.gameOver()) {
+      return 'lose';
+    }
+
+    return 'playing';
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.genereteCell();
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.isStart = true;
+
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 4; j++) {
+        this.field[i][j] = 0;
+      }
+    }
+
+    this.genereteCell();
+
+    this.score = 0;
+    this.getScore();
+  }
 
   // Add your own methods here
+  genereteCell() {
+    const emptyCells = [];
+
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 4; j++) {
+        if (this.field[i][j] === 0) {
+          emptyCells.push({ x: i, y: j });
+        }
+      }
+    }
+
+    if (emptyCells.length > 0) {
+      const { x, y } =
+        emptyCells[Math.floor(Math.random() * emptyCells.length)];
+
+      if (this.field[x][y] === 0) {
+        this.field[x][y] = Math.random() < 0.9 ? 2 : 4;
+
+        if (this.isStart === true) {
+          const z = Math.floor(Math.random() * 3) + 1;
+
+          if (x + z <= 3 && y + z <= 3) {
+            this.field[x + z][y + z] = Math.random() < 0.9 ? 2 : 4;
+          }
+
+          this.isStart = false;
+        }
+      }
+    } else {
+      this.getStatus();
```

Comment:
this function has the wrong name or does the wrong thing. Get status method should only return status